source/Irrlicht/CMakeLists.txt

```diff
@@ -529,11 +537,27 @@ target_include_directories(IrrlichtMt
 		"$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include/>"
 		"$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>"
 		"$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/irrlichtmt>"
-	PRIVATE
-		${link_includes}
 )
 
-target_link_libraries(IrrlichtMt PRIVATE ${link_libs})
+target_link_libraries(IrrlichtMt
+	PRIVATE
+		# FIXME this will not propogate dependencies to IrrlichtMt
+		# dependants, but I do not want to clutter the export with
+		# object libraries, either.
+		<$BUILD_INTERFACE:IRROBJ>
+		<$BUILD_INTERFACE:IRROTHEROBJ>
+		<$BUILD_INTERFACE:IRRVIDEOOBJ>
```

Comment:
This is not a portable solution. I will probably end up including these in the export for the time being.